site/book/05-developing-functions/02-developing-in-Go.md

```diff
@@ -13,121 +12,161 @@ the following features:
+func Run()(rl *fn.ResourceList) (bool, error) {
+// Your code
+}
+
 func main() {
-	// create a struct matching the structure of ResourceList.FunctionConfig to hold its data
-	var config struct {
-		Data map[string]string `yaml:"data"`
-	}
-	fn := func(items []*yaml.RNode) ([]*yaml.RNode, error) {
-		for i := range items {
-			// set the annotation on each resource item
-			err := items[i].PipeE(yaml.SetAnnotation("myannotation", config.Data["myannotation"]))
-			if err != nil {
-				return nil, err
-			}
-		}
-		return items, nil
-	}
-	p := framework.SimpleProcessor{Filter: kio.FilterFunc(fn), Config: &config}
-	cmd := command.Build(p, command.StandaloneDisabled, false)
-  // Adds a "gen" subcommand to create a Dockerfile for building the function into a container image.
-  command.AddGenerateDockerfile(cmd)
-
-	if err := cmd.Execute(); err != nil {
-		os.Exit(1)
-	}
+  // CUSTOMIZE IF NEEDED
+  // `AsMain` accepts a `ResourceListProcessor` interface.
+  // You can explore other `ResourceListProcessor` structs in the SDK or define your own.
+  if err := fn.AsMain(fn.ResourceListProcessorFunc(Run)); err != nil {
+    os.Exit(1)
+  }
 }
 ```
 
-### Build and test the function
+The `fn` library provides a series of KRM level operations for `ResourceList`. 
+Basically, the KRM resource `ResourceList.FunctionConfig` and KRM resources `ResourceList.Items` are both converted to 
+`KubeObject` objects. You can use `KubeObject` similar as [`unstructured.Unstrucutred`](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured).
 
-Build the go binary:
+To learn more about the `KubeObject`, see https://pkg.go.dev/github.com/GoogleContainerTools/kpt-functions-sdk/go/fn
 
-```shell
-$ go mod tidy; go build -o my-fn .
-```
-
-Fetch the wordpress package:
 
-```shell
-$ kpt pkg get https://github.com/GoogleContainerTools/kpt.git/package-examples/wordpress@v0.8
-```
+### Test the KRM function
 
-Test it by running the function imperatively:
+The "get-started" package contains a `./data` directory. You can use this to test out your functions. 
 
 ```shell
-$ kpt fn eval wordpress --exec ./my-fn -- myannotation=foo
-```
+# Edit the `data/resources.yaml` with your KRM resources. 
+# resources.yaml already has a `Deployment` and `Service` as test data. 
+vim data/resources.yaml
 
-During iterative development, `--exec` flag can be used to execute the
-function binary directly instead of requiring the function to be containerized
-first. Once you have a function binary that works, you can then proceed to
-creating the container image.
+# Edit `ResourceList.FunctionConfig` with your own function config. 
+# You can start with a `ConfigMap` as FunctionConfig. 
+# You can skip this step if your KRM function does not require `ResourceList.FunctionConfig`. 
+vim data/fn-config.yaml
 
-### Publish the function
+# Convert the KRM resources and FunctionConfig resource to `ResourceList`.
+kpt fn source data --fn-config data/fn-config.yaml
```

Comment:
Another option is that include a ResourceList resource in the get-started package.

Reply:
I'm wondering if that will be easier to understand since users still need a way to pipe the ResourceList as a STDIN, and it will be harder if they want to use their own input resource.  WDYT?